tests/integration/test_httpx.py

```diff
@@ -238,6 +238,37 @@ def test_behind_proxy(do_request):
         assert cassette_response.request.url == response.request.url
 
 
+def test_auth_flow(do_request):
+    yml = os.path.dirname(os.path.realpath(__file__)) + "/cassettes/" + "test_httpx_test_test_auth_flow.yml"
+    url = "https://mockbin.org/headers"
+    client = None
+
+    class AuthWithExtraRequest(httpx.Auth):
+        def auth_flow(self, request):
+            response = yield client.client.build_request("GET", url, params={"foo": "bar"})
+            assert response
+            request.headers["Foobar"] = "somethingsomething"
+            yield request
+
+    with vcr.use_cassette(yml):
+        with do_request() as client:
+            client("GET", url, auth=AuthWithExtraRequest())
```

Comment:
The default value for `follow_redirects` is `False` on httpx client.
I think if you call the client with `client("GET", url, auth=AuthWithExtraRequest(), follow_redirects=True)` we don't need to overwrite a private method.

Reply:
I don't really understand what you wrote. If you think something else works please try it.